### PR TITLE
feat: implement GET /collections/{collection_id}/versions

### PR DESCRIPTION
**Pull request by danieljhegeman**
## Reason for Change

- #4328 

## Changes

- add: collections 'versions' endpoint
- modify: 
  - consolidate "identifier is not uuid format" error-checking code into common function and replace in existing code. 
  - Standardize on 403 rather than some 404 and some 403.
  - unrelated typo fix for function name

## Testing steps

- Unit tested

### CODE REVIEW

File: backend/curation/api/v1/curation/collections/collection_id/versions.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+from flask import jsonify, make_response
+
+from backend.curation.api.v1.curation.collections.common import reshape_for_curation_api
+from backend.layers.common.entities import CollectionId
+from backend.portal.api.providers import get_business_logic
+
+
+def get(collection_id: str):
```
Comment (nayib-jose-gloria) [inline backend/curation/api/v1/curation/collections/collection_id/versions.py:8]: we should add a check that the collection_id is a UUID and return an error code (404 or 403, whatever we end up deciding is appropriate) if not.

File: backend/curation/api/v1/curation/collections/common.py [hunk 2]
```diff
@@ -65,7 +65,7 @@ def reshape_for_curation_api(
         collection_id = collection_version.collection_id
         collection_url = f"{get_collections_base_url()}/collections/{collection_id.id}"
         revision_of = None
-        if not user_info.is_user_owner_or_allowed(collection_version.owner):
+        if not user_info or user_info.is_user_owner_or_allowed(collection_version.owner):
```
Comment (nayib-jose-gloria) [inline backend/curation/api/v1/curation/collections/common.py:68]: should be` if not user_info or not user_info.user_info.is_user_owner_or_allowed(collection_version.owner)`

File: backend/curation/api/v1/curation/collections/collection_id/versions.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    Return all published versions for a Collection
+    """
+    collection_versions = list(
+        map(
+            reshape_for_curation_api,
+            get_business_logic().get_all_published_collection_versions_from_canonical(CollectionId(collection_id)),
+        )
+    )
+
+    collection_versions.sort(key=lambda c: c["published_at"])
```
[the remark this answers is not in the document]
Comment (danieljhegeman) [reply backend/curation/api/v1/curation/collections/collection_id/versions.py]: We'll go 404 across the board, [as supported by Andrew](https://czi-sci.slack.com/archives/C02UGUV5R3J/p1679072261120489?thread_ts=1678908526.729839&cid=C02UGUV5R3J).

File: backend/curation/api/v1/curation/collections/collection_id/versions.py (hunk 1 above)
Comment (danieljhegeman) [reply backend/curation/api/v1/curation/collections/collection_id/versions.py:8]: I've seen that you put that in in a few places, though it looks like a mix of 403 and 404.

I'm going to 
1) factor out this code into `common.py`, and 
2) standardize on 403

File: backend/curation/api/v1/curation/collections/collection_id/actions.py [hunk 4]
```diff
@@ -5,7 +5,7 @@
 from backend.common.utils.http_exceptions import InvalidParametersHTTPException, MethodNotAllowedException
 from backend.curation.api.v1.curation.collections.common import (
     extract_doi_from_links,
-    get_infered_collection_version_else_forbidden,
+    get_inferred_collection_version_else_forbidden,
```
Comment (danieljhegeman) [inline backend/curation/api/v1/curation/collections/collection_id/actions.py:8]: unrelated change

**Review by nayib-jose-gloria — APPROVED** [review]
